x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/__mocks__/request_responses.ts

```diff
@@ -224,69 +245,91 @@ export const getFindResultWithMultiHits = ({
-  },
-  payload,
-});
-
-export const getImportRulesRequestOverwriteTrue = (payload?: Buffer): ServerInjectOptions => ({
-  method: 'POST',
-  url: `${DETECTION_ENGINE_RULES_URL}/_import?overwrite=true`,
-  headers: {
-    'Content-Type': `multipart/form-data; boundary=${TEST_BOUNDARY}`,
-  },
-  payload,
-});
-
-export const getDeleteRequest = (): ServerInjectOptions => ({
-  method: 'DELETE',
-  url: `${DETECTION_ENGINE_RULES_URL}?rule_id=rule-1`,
-});
-
-export const getDeleteRequestById = (): ServerInjectOptions => ({
-  method: 'DELETE',
-  url: `${DETECTION_ENGINE_RULES_URL}?id=04128c15-0d1b-4716-a4c5-46997ac7f3bd`,
-});
-
-export const getCreateRequest = (): ServerInjectOptions => ({
-  method: 'POST',
-  url: DETECTION_ENGINE_RULES_URL,
-  payload: {
-    ...typicalPayload(),
-  },
-});
-
-export const getSetSignalStatusByIdsRequest = (): ServerInjectOptions => ({
-  method: 'POST',
-  url: DETECTION_ENGINE_SIGNALS_STATUS_URL,
-  payload: {
-    ...typicalSetStatusSignalByIdsPayload(),
-  },
-});
-
-export const getSetSignalStatusByQueryRequest = (): ServerInjectOptions => ({
-  method: 'POST',
-  url: DETECTION_ENGINE_SIGNALS_STATUS_URL,
-  payload: {
-    ...typicalSetStatusSignalByQueryPayload(),
-  },
-});
-
-export const getSignalsQueryRequest = (): ServerInjectOptions => ({
-  method: 'POST',
-  url: DETECTION_ENGINE_QUERY_SIGNALS_URL,
-  payload: { ...typicalSignalsQuery() },
-});
-
-export const getSignalsAggsQueryRequest = (): ServerInjectOptions => ({
-  method: 'POST',
-  url: DETECTION_ENGINE_QUERY_SIGNALS_URL,
-  payload: { ...typicalSignalsQueryAggs() },
-});
+export const ruleStatusRequest = () =>
+  requestMock.create({
+    method: 'get',
+    path: `${DETECTION_ENGINE_RULES_URL}/_find_statuses`,
+    query: { ids: ['someId'] },
+  });
+
+export const getImportRulesRequest = (hapiStream?: HapiReadableStream) =>
+  requestMock.create({
+    method: 'post',
+    path: `${DETECTION_ENGINE_RULES_URL}/_import`,
+    headers: {
+      'Content-Type': 'multipart/form-data; boundary=???',
+    },
+    body: { file: hapiStream },
+  });
+
+export const getImportRulesRequestOverwriteTrue = (hapiStream?: HapiReadableStream) =>
+  requestMock.create({
+    method: 'post',
+    path: `${DETECTION_ENGINE_RULES_URL}/_import`,
+    headers: {
+      'Content-Type': 'multipart/form-data; boundary=???',
+    },
+    body: { file: hapiStream },
+    query: { overwrite: true },
+  });
+
+export const getDeleteRequest = () =>
+  requestMock.create({
+    method: 'delete',
+    path: DETECTION_ENGINE_RULES_URL,
+    query: { rule_id: 'rule-1' },
+  });
+
+export const getDeleteRequestById = () =>
+  requestMock.create({
+    method: 'delete',
+    path: DETECTION_ENGINE_RULES_URL,
+    query: { id: '04128c15-0d1b-4716-a4c5-46997ac7f3bd' },
+  });
+
+export const getCreateRequest = () =>
+  requestMock.create({
+    method: 'post',
+    path: DETECTION_ENGINE_RULES_URL,
+    body: {
+      ...typicalPayload(),
```

Comment:
I don't know at what point we did spreads when they're not needed. This is might be an existing mistake but in these cases it should just be:

```ts
body: typicalPayload(),
```